docs/ingestion/migrate-from-firehose-ingestion.md

```diff
@@ -0,0 +1,196 @@
+---
+id: migrate-from-firehose
+title: "Migrate from firehose to input source ingestion"
+sidebar_label: "Migrate from firehose"
+---
+
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+Apache deprecated support for Druid firehoses in version 0.17. Support for firehose ingestion was removed in LTS version 0.24.
+
+If you're using a firehose for batch ingestion, we strongly recommend that you follow the instructions on this page to transition to using native batch ingestion input sources as soon as possible. 
+
+Firehose ingestion doesn't work with newer Druid versions, so you must be using an ingestion spec with a defined input source before you upgrade. 
+
+## Migrate from firehose ingestion to an input source
+
+To migrate from firehose ingestion, copy your existing ingestion spec into a new file. Refer to [Native batch ingestion with firehose (Deprecated)](./native-batch-firehose.md) for a description of firehose properties.
+
+Edit the new file as follows:
+
```

Comment:
It should be noted, that the web console has a way of doing this conversion for you. It literally does the steps you outlined below.

In fact, if you add the `"type": "string"` (which I, and the console believe is required) to your "before" config below and then copy paster it into the `Edit spec` stage of the data loader.

<img width="1363" alt="image" src="https://user-images.githubusercontent.com/177816/190516498-57955af9-9813-4c8a-8914-7e409df29f83.png">

It will be instantly converted to the "after" spec you have below.

My recommendation to a user would be to use the console tool first and if it fails for some reason to use the steps you outlined. Note that the console tool only works for the `string` parser.

Reply:
@tejaswini-imply can you please review, test and comment?

Reply:
Added the `"type": "string"` property to the firehose spec as outlined by Vad and updated the instructions to include the console option. @vogievetsky @tejaswini-imply please review.